server/controllers/v1/auth.controller.js

```diff
@@ -64,23 +64,50 @@ const resetPassword = async (request, response) => {
   const token = request.body.token;
   const existingPasswordReset = await PasswordReset.findOne({ token });
   if (existingPasswordReset) {
-    await User.findOneAndUpdate(
-      {
-        email: existingPasswordReset.email
-      },
-      {
-        password: BCrypt.hashSync(request.body.password)
-      }
-    );
-    await PasswordReset.findOneAndDelete({
-      email: existingPasswordReset.email
+    const timeInMinutes = getResetLinkLifetime(existingPasswordReset);
```

Comment:
Codacy found an issue: ['getResetLinkLifetime' was used before it was defined.](https://app.codacy.com/manual/sam-spain/full-stack-js/pullRequest?prid=6184982)